client/filter/tvs.geneVariant.js

```diff
@@ -21,235 +24,267 @@ export const handler = {
 }
 
 async function fillMenu(self, _div, tvs) {
-	const data = await self.opts.vocabApi.getCategories(tvs.term, self.filter, {})
-	const dtClassMap = new Map()
-	for (const l of data.lst) {
-		dtClassMap.set(l.dt, l.classes)
-	}
-	const div = _div.append('div').style('padding', '5px').style('cursor', 'pointer')
+	const term = structuredClone(tvs.term)
+	// add a variant filter onto the geneVariant term
+	await mayMakeVariantFilter({ q: {}, term }, self.opts.vocabApi)
+	if (!term.filter?.terms?.length) return
+	// generate a frontend vocab using dt terms from the variant filter
+	// and render a data dictionary of this vocab
+	const termdb = await import('../termdb/app')
+	termdb.appInit({
+		holder: _div.append('div').style('margin-top', '15px'),
+		state: {
+			nav: {
+				header_mode: 'hide_search'
+			},
+			vocab: {
+				// no need to pass vocabApi as a termdb.appInit() option, since
+				// termdb app will create vocabApi as needed from state.vocab
+				// if provided
+				terms: term.filter.terms,
+				parent_termdbConfig: self.opts.vocabApi.termdbConfig
+			},
+			tree: { usecase: { target: 'filter' } }
+		},
+		tree: {
+			click_term2select_tvs(tvs) {
+				tvs.term.geneVariantTerm = structuredClone(term) // track the parent geneVariant term
+				self.opts.callback(tvs)
+			}
+		}
+	})
 
-	div.append('div').style('font-size', '1.2rem').text(self.tvs.term.name)
+	return
```

Comment:
@gavrielm  I think the commented out code should be deleted. I had kept this earlier in case it may be needed for something else, but it doesn't look like it'll be used again.